app/models/course.rb

```diff
@@ -97,7 +97,7 @@ class Course < ApplicationRecord
              optional: true
 
   has_many :course_subjects
-  has_many :subjects, through: :course_subjects
+  has_many :subjects, -> { order "course_subject.priority" }, through: :course_subjects
```

Comment:
I'd actually move the ordering into `course_subjects` and add `before_add` hook to automatically order any assigned `CourseSubject` objects accordingly:

```ruby
  has_many :course_subjects,
           -> { order :priority },
           inverse_of: "course",
           before_add: :set_course_subject_position

  def set_course_subject_position(course_subject)
    return if course_subject.priority.present?

    course_subject.priority = if course_subjects.any?
                                course_subjects.last.priority + 1
                              else
                                0
                              end
  end
```

Reply:
This means that we would can add subjects to a course and get ordering by default:

```ruby
> c = Course.new
=> #<Course:0x00007fcb9c85b398 ...>
> english = Subject.find_by(subject_name: "English")
=> #<SecondarySubject:0x00007fcb9a5bab18 id: 21, ...>
> maths = Subject.find_by(subject_name: "Mathematics")
=> #<SecondarySubject:0x00007fcb9d4a4208 id: 25, ...>
> c.subjects = [maths, english]
=> [#<SecondarySubject:0x00007fcb9a6107c0 id: 25, ...>,
 #<SecondarySubject:0x00007fcb9a4f9cd8 id: 21, ...>]
> c.course_subjects
=> [#<CourseSubject:0x00007fcb9d1d6d00 id: nil, course_id: nil, subject_id: 25, created_at: nil, updated_at: nil, priority: 0>,
 #<CourseSubject:0x00007fcb9d234d88 id: nil, course_id: nil, subject_id: 21, created_at: nil, updated_at: nil, priority: 1>]
```

This even appears to work when a course already has a subject, the priorities end up starting at 0:

```ruby
[8] pry(main)> c = rc.courses.sample
=> #<Course:0x00007fc5dea20bd8
 id: 12924670,
 ...>
[9] pry(main)> c.subjects
=> [#<SecondarySubject:0x00007fc5c2148388 id: 28, type: "SecondarySubject", subject_code: "C6", subject_name: "Physical education", created_at: nil, updated_at: nil, subject_area_id: nil>]
[10] pry(main)> c.course_subjects
=> [#<CourseSubject:0x00007fc5c2218e70 id: 25900, course_id: 12924670, subject_id: 28, created_at: nil, updated_at: nil, priority: nil>]
[11] pry(main)> c.course_subjects.first.priority = 1
=> 1
[12] pry(main)> c.course_subjects.first.save
=> true
[13] pry(main)> c.course_subjects
=> [#<CourseSubject:0x00007fc5c2218e70 id: 25900, course_id: 12924670, subject_id: 28, created_at: nil, updated_at: Tue, 11 Feb 2020 19:11:08 UTC +00:00, priority: 1>]
[14] pry(main)> c.subjects = [maths, english]
=> [#<SecondarySubject:0x00007fc5de700e30 id: 25, type: "SecondarySubject", subject_code: "G1", subject_name: "Mathematics", created_at: nil, updated_at: nil, subject_area_id: nil>,
 #<SecondarySubject:0x00007fc5da160ba0 id: 21, type: "SecondarySubject", subject_code: "Q3", subject_name: "English", created_at: nil, updated_at: nil, subject_area_id: nil>]
[15] pry(main)> c.course_subjects
=> [#<CourseSubject:0x00007fc5de90b0e0 id: 40785, course_id: 12924670, subject_id: 25, created_at: Tue, 11 Feb 2020 19:11:48 UTC +00:00, updated_at: Tue, 11 Feb 2020 19:11:48 UTC +00:00, priority: 0>,
 #<CourseSubject:0x00007fc5de372048 id: 40786, course_id: 12924670, subject_id: 21, created_at: Tue, 11 Feb 2020 19:11:48 UTC +00:00, updated_at: Tue, 11 Feb 2020 19:11:48 UTC +00:00, priority: 1>]
```

Reply:
btw, one reason I think it's more appropriate to order `course_association` is because the `priority` column exists there.